File: python/cash.py
# prints out the least number of coins that can be returned equal to the change owed
# use the get_float method from cs50 library
from cs50 import get_float

# expects input to be 9.00, 0.45, .2, but not 912 or $9.12. range is more than 0 and less than $10
while True:
    pennies = get_float("Change owed: ")
    if pennies > 0 and pennies < 1000:
        break

# easier to deal with whole numbers 192 cents rather than $1.92 or 1.92 float dollars
pennies = int(pennies * 100)
# keep track of the total coins dispensed
coins = 0

while True:
    if (pennies - 25) >= 0:
        coins += 1
        pennies -= 25
    else:
        break

while True:
    if (pennies - 10) >= 0:
        coins += 1
        pennies -= 10
    else:
        break

while True:
    if (pennies - 5) >= 0:
        coins += 1
        pennies -= 5
    else:
        break

coins = coins + pennies
print(coins)

File: c/cash.c
#import <stdio.h>
#import <cs50.h>

int main(void)
{
    float cash = 0;
    do
    {
        cash = get_float("How much change we talking (0.01-1000)? ");
    }
    while (cash <= 0 || cash > 1000); // reasonable boundary 0 < x < 1000
    // multiply by 100 so I can divide below by whole numbers
    // i.e.: $1.52 => 152. 152 - 25 cents instead of 1.52 - 0.25 cents
    // no real reason to do this other than I think it reads easier
    cash = cash*100;
        
    // track how many coins have been dispensed
    int coins = 0;
       
    // while able to remove 25 cents from the current cash total...
    while (cash - 25 >= 0)
    {
        // lower the total by 25 cents
        cash -= 25;
        // increment the coin counter by +1 coin
        coins++;
    }
    // repease for dimes and nickels
    while (cash - 10 >= 0) 
    {
        cash -= 10;
        coins++;
    }
    while (cash - 5 >= 0)
    {
        cash -= 5;
        coins++;
    }
    
    // all that is left at this point is pennies (total is under 5c)
    // convert the float to an int (removing all decimals) leaving whole pennies
    int int_cash = cash;
    // add the pennies to the coins counter
    coins += int_cash;
    // simple print function returns the coin counter value    
    printf("%i\n", coins);
}
